### PR TITLE
feat(jtk): fix remaining output compliance gaps

### LINKED ISSUE #312: feat(jtk): fix remaining output compliance gaps (issues list sprint, issues fields complex values, gadgets type, show-components tree)
Four remaining output compliance gaps found during full-surface re-audit against #230.

---

### 1. `issues list --extended` — SPRINT column not resolving

The `SPRINT` column shows `-` for all issues in `issues list --extended`, even when the issue has a sprint. The same sprint-from-`customfield_10020` fix that was applied to `issues get --extended` was not carried over to the list path. `sprints issues --extended` (different code path) resolves the sprint correctly for the same issues.

**Before:**
```
$ bin/jtk issues search --jql "issue = MON-1918" --extended
KEY | STATUS | TYPE | PTS | ASSIGNEE | REPORTER | SPRINT | PARENT | UPDATED | LABELS | COMPONENTS | SUMMARY
MON-1918 | In Development | SDLC | 5 | Devin Quirk | Rusty Hall | - | MON-3859 | 2026-04-10 | - | - | Q2 and Banno Notifications for MSR send
```

**After:**
```
KEY | STATUS | TYPE | PTS | ASSIGNEE | REPORTER | SPRINT | PARENT | UPDATED | LABELS | COMPONENTS | SUMMARY
MON-1918 | In Development | SDLC | 5 | Devin Quirk | Rusty Hall | MON Sprint 70 | MON-3859 | 2026-04-10 | - | - | Q2 and Banno Notifications for MSR send
```

Cross-check: `bin/jtk sprints issues 125 --extended | grep MON-1918` already shows `MON Sprint 70` in the SPRINT column — the data is there, the list path just doesn't resolve it.

---

### 2. `issues fields` — complex field values render as raw Go maps

Simple fields (`string`, `number`, `option`, `user`) render cleanly. Fields with complex/nested types dump raw Go representations into the VALUE column, making output unreadable.

**Before:**
```
$ bin/jtk issues fields MON-4970
FIELD_ID | NAME | TYPE | VALUE
aggregateprogress | Σ Progress | progress | map[progress:0 total:0]
comment | Comment | comments-page | map[comments:[map[author:map[accountId:61292e4c... (hundreds of characters)
customfield_10000 | Development | any | {pullrequest={dataType=pullrequest, state=MERGED, stateCount=1}, json={"cachedValue":... (hundreds of characters)
```

**After:** Fields whose VALUE cannot be rendered as a clean scalar should be omitted from the output (same principle as Labels/Components in `issues get` — only shown when non-empty and renderable). The `FIELD_ID`, `NAME`, and `TYPE` columns should only appear for fields with a printable value.

---

### 3. `dashboards gadgets list` — TYPE column always empty

The gadget TYPE is not returned as a dedicated field by the Jira API. It is embedded in the `uri` field returned per gadget, e.g.:

```
rest/gadgets/1.0/g/com.atlassian.jira.gadgets:assigned-to-me-gadget/gadgets/assigned-to-me-gadget.xml
```

The code needs to extract the type identifier from the URI (the segment between the last `:` and the next `/`).

**Before:**
```
$ bin/jtk dashboards gadgets list 10000
ID | POSITION | TITLE | TYPE
10001 | 1,1 | Spaces | 
10002 | 0,1 | Assigned to Me | 
10054 | 0,0 | Filter Results | 
```

**After:**
```
ID | POSITION | TITLE | TYPE
10001 | 1,1 | Spaces | project-gadget
10002 | 0,1 | Assigned to Me | assigned-to-me-gadget
10054 | 0,0 | Filter Results | filter-results-gadget
```

---

### 4. `automation get --show-components` — flat table instead of indented tree

Spec says: "dumps the full component tree as indented text (trigger → conditions → actions)". Current output is a flat numbered `# | COMPONENT | TYPE` table with no hierarchy.

**Before:**
```
$ bin/jtk automation get 018c2840-57c1-7869-9393-11205cc87ce4 --show-components
018c2840-57c1-7869-9393-11205cc87ce4  ON/MON: Create Onboarding Tasks
State: ENABLED
Components: 27 total — 4 conditions, 23 actions
Description: Creates Tasks when a new Onboarding Epic is created
# | COMPONENT | TYPE
1 | CONDITION | jira.jql.condition
2 | ACTION | jira.create.variable
...
```

**After:** Indented text with hierarchy visible, trigger first, then conditions and actions beneath:
```
018c2840-57c1-7869-9393-11205cc87ce4  ON/MON: Create Onboarding Tasks
State: ENABLED
Components: 27 total — 4 conditions, 23 actions
Description: Creates Tasks when a new Onboarding Epic is created

TRIGGER  jira.issue.event.trigger:created
  CONDITION  jira.jql.condition
  ACTION  jira.create.variable
  ACTION  jira.create.variable
  ...
```

---

### Spec/implementation note (no code change needed — spec should be updated)

`issues field-options` takes `<issue-key> <field>` in the implementation but the spec example shows `jtk issues field-options MON customfield_10050` (project key). Passing a project key fails. The implementation's approach (issue key for project-context resolution) is correct — the spec example should be updated to use an issue key like `jtk issues field-options MON-4970 customfield_10050`.

---

## Success criteria

Build and lint before testing:
```
make build-jtk && make lint
```

**1. `issues list --extended` sprint resolves.**
```
bin/jtk issues search --jql "issue = MON-1918" --extended
```
The `SPRINT` column must show `MON Sprint 70`, not `-`. Verify it matches:
```
bin/jtk sprints issues 125 --extended | grep MON-1918
```
Both outputs must show the same sprint value.

**2. `issues fields` omits unrenderable complex fields.**
```
bin/jtk issues fields MON-4970
```
The `comment`, `aggregateprogress`, and `customfield_10000` (Development) rows must not appear. All remaining rows must have a VALUE that is a clean scalar string (no `map[...]` or `{...}` raw representations).

**3. `dashboards gadgets list` TYPE column is populated.**
```
bin/jtk dashboards gadgets list 10000
```
The TYPE column must be non-empty for each gadget. Expected values based on live data:
- Spaces → `project-gadget`
- Assigned to Me → `assigned-to-me-gadget`
- Filter Results → `filter-results-gadget`

**4. `automation get --show-components` renders indented tree.**
```
bin/jtk automation get 018c2840-57c1-7869-9393-11205cc87ce4 --show-components
```
Output must not contain a numbered table with `# | COMPONENT | TYPE` header. It must show an indented hierarchy with the TRIGGER at the top level and CONDITIONs/ACTIONs indented beneath it.

### PR DESCRIPTION
**Pull request by rianjs**
## Summary

- **Sprint column**: Add `customfield_10020` to `DefaultSearchFields` and SPRINT column spec `Fetch` so `issues list --extended` resolves sprint names
- **Complex field values**: Return empty string for unrenderable complex types in `FormatCustomFieldValue()` — fields are omitted from output instead of showing raw `map[...]`
- **Gadgets TYPE**: Extract gadget type from URI field as fallback when `ModuleID` is empty
- **Show-components tree**: Replace flat `# | COMPONENT | TYPE` table with indented tree that recursively renders trigger, conditions, and children

## Test plan

- [x] `make build` — compiles
- [x] `make test` — all tests pass
- [x] `make lint` — zero issues
- [ ] Live: `bin/jtk issues search --jql "issue = MON-1918" --extended` — SPRINT shows `MON Sprint 70`
- [ ] Live: `bin/jtk issues fields MON-4970` — no `map[...]` values
- [ ] Live: `bin/jtk dashboards gadgets list 10000` — TYPE column populated
- [ ] Live: `bin/jtk automation get 018c2840-57c1-7869-9393-11205cc87ce4 --show-components` — indented tree

Closes #312

### CODE REVIEW

**Comment by rianjs** [comment]
## TDD Coverage Review

### What's well-covered

- **`gadgetTypeFromURI`**: Table-driven tests hit standard URI, no-colon, no-slash, and empty string cases. The `TestPresentGadgets_URIFallback` integration test validates the ModuleID-vs-URI fallback priority at the presentation layer. Good.
- **`FormatCustomFieldValue` fallback paths**: Two new cases (`unhandled_map`, `unhandled_type`) cover the `""` return change. Adequate.
- **Component tree rendering**: `TestAutomationPresenter_PresentGetDetail_ShowComponents` updated to assert indentation and type text rather than just section type. The `get_test.go` integration assertion likewise checks formatted output. Acceptable for a rendering function.

### Gaps and flags

**`DecodedChildren` / `DecodedConditions` — no unit tests at all.**
These are the recursive building blocks of `renderComponent`. There are no tests for:
- Valid JSON round-trips (`Children` / `Conditions` set to valid JSON arrays)
- Malformed JSON (should return `nil` per current code — but this is unverified)
- Empty `json.RawMessage` vs `nil` (both return `nil`; worth a quick assertion)

**`renderComponent` recursion — not tested.**
`renderComponent` calls itself for `DecodedConditions()` and `DecodedChildren()`. There is no test that exercises a multi-depth tree (e.g., TRIGGER → ACTION → nested CONDITION). The only tested shape is a flat TRIGGER + one ACTION. A two-level nested component would catch off-by-one indent bugs immediately.

**`componentTree` with `nil` trigger and empty components.**
The guard `rule.Trigger != nil || len(rule.Components) > 0` means `componentTree` can be called with both nil and empty slice. Calling it in that state produces an empty `MessageSection` — that's probably fine, but it's not asserted.

**`componentTree` TRIGGER deduplication logic.**
The code skips a `TRIGGER` component from `rule.Components` if `trigger != nil`. There's no test for the case where `rule.Trigger` is set *and* a duplicate TRIGGER also appears in `rule.Components` — which is presumably the real-world shape this is guarding against.

**Sprint field alias (`customfield_10020`) — no test.**
Adding an alias/fetch field to `IssueListSpec` is a one-liner, but there's no test asserting that the sprint column resolves via `customfield_10020` when `sprint` is absent. If the projection registry does alias resolution, a test here would catch a regression if the field ID changes again.

### Minor

- `gadgetTypeFromURI` is unexported and tested directly from within `package present`. That's fine. No issue.
- The silent discard of unmarshal errors in `DecodedChildren`/`DecodedConditions` is an intentional design choice (returns `nil` on bad JSON), but with no test it's easy to accidentally change the behaviour.

### Summary

The happy-path coverage is solid. The critical gap is the absence of any tests for `DecodedChildren`/`DecodedConditions` and for recursive/nested `renderComponent` trees. Everything else is low risk. Recommend adding at minimum: unit tests for the two `Decoded*` methods (valid JSON, bad JSON, nil) and one nested-tree test for `renderComponent` before merging.

**Comment by rianjs** [comment]
Addressing TDD findings:

1. **DecodedChildren/DecodedConditions** — Added: unit tests for valid JSON round-trip, nil/empty, and malformed JSON.
2. **Recursive renderComponent** — Added: `TestComponentTree_NestedChildren` exercises 3-level nesting (CONDITION → CONDITION_BLOCK → ACTION).
3. **TRIGGER deduplication** — Added: `TestComponentTree_TriggerDeduplication` verifies only 1 TRIGGER line when present in both `rule.Trigger` and `rule.Components`.
4. **Sprint alias test** — Accepted risk. The alias is exercised through `Registry.Match()` which has its own tests. Adding another test for this specific alias would test the framework, not the feature.

**Comment by rianjs** [comment]
Note: Codex PR review returned stale plan-phase response (all findings were already addressed in implementation: Conditions decoded, alias added, trigger-aware guard). Proceeding to daemon review.

**Comment by rianjs** [comment]
**Blocker**
- None.

**Major**
- `tools/jtk/internal/present/automation.go`: `PresentDetail(rule, true)` still renders the old flat `# | COMPONENT | TYPE` table inline. That leaves two component-detail contracts in the presenter and keeps non-compliant behavior alive. Route this path through `componentTree` too, or remove the `showComponents` branch if only `automation get` supports it.

**Minor**
- `tools/jtk/internal/present/automation_test.go`: add coverage for `DecodedConditions()`, not just nested `children`. Use a `CONDITION_BLOCK` with a comparator in `conditions` plus an action in `children` to lock the intended tree order.
- Add a projection/fetch test for `--extended --fields SPRINT` and `--extended --fields customfield_10020` deriving `customfield_10020`; this was the subtle sprint architecture gap.

**Nit**
- None.

File: tools/jtk/api/issues.go [hunk 1]
```diff
@@ -413,7 +413,7 @@ func FormatCustomFieldValue(v any) string {
 		if s, ok := val["displayName"].(string); ok {
 			return s
 		}
-		return fmt.Sprintf("%v", val)
+		return ""
```
Comment (monit-reviewer) [inline tools/jtk/api/issues.go:419]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): FormatCustomFieldValue now returns "" for any map[string]any that doesn't match the four known keys and for all unrecognized types. This silently drops field data with no way for users to distinguish an empty field from one whose type is unhandled. The PR intentionally removes noisy map[...] output, but a sentinel like "(unsupported)" or debug-level logging would preserve debuggability when users encounter missing field values.

*Reply to this thread when addressed.*

File: tools/jtk/api/automation_types.go [hunk 2]
```diff
@@ -76,6 +76,30 @@ type RuleComponent struct {
 	ConnectionID  string          `json:"connectionId,omitempty"`
 }
 
+// DecodedChildren decodes the Children json.RawMessage into a slice of RuleComponent.
+func (c *RuleComponent) DecodedChildren() []RuleComponent {
+	if len(c.Children) == 0 {
+		return nil
```
Comment (monit-reviewer) [inline tools/jtk/api/automation_types.go:82]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): DecodedChildren and DecodedConditions silently swallow JSON unmarshal errors by returning nil. A malformed payload from the Jira API will produce a silently truncated component tree with no signal to the caller or user. Consider returning an error or logging a warning so callers can distinguish 'no children' from 'invalid JSON in children payload'.

*Reply to this thread when addressed.*

File: tools/jtk/internal/present/automation.go [hunk 3]
```diff
@@ -299,23 +299,42 @@ func (AutomationPresenter) PresentGetDetailExtended(rule *api.AutomationRule, sh
...
 	return &present.OutputModel{Sections: sections}
 }
 
-func componentTable(components []api.RuleComponent) *present.TableSection {
-	rows := make([]present.Row, len(components))
-	for i, c := range components {
-		rows[i] = present.Row{
-			Cells: []string{fmt.Sprintf("%d", i+1), c.Component, c.Type},
+func componentTree(trigger *api.RuleComponent, components []api.RuleComponent) *present.MessageSection {
+	var lines []string
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/automation.go:310]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): componentTree renders all non-TRIGGER entries from the flat components slice at a fixed depth of 1, implicitly assuming they are all direct children of the trigger. If the API ever returns root-level components that are siblings of the trigger (e.g., an error-handler), they will appear incorrectly indented. The assumption is undocumented and untested against real API responses with non-standard component layouts.

*Reply to this thread when addressed.*

File: tools/jtk/internal/present/automation.go [hunk 4]
```diff
@@ -299,23 +299,42 @@ func (AutomationPresenter) PresentGetDetailExtended(rule *api.AutomationRule, sh
...
+	if trigger != nil {
+		renderComponent(&lines, trigger, 0)
+	}
+	for i := range components {
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/automation.go:314]: 🔵 **Low** (harness-engineering:harness-enforcement-reviewer): TRIGGER deduplication skips all TRIGGER-typed entries in components when rule.Trigger != nil, regardless of whether they represent the same or a different trigger. If a rule legitimately has a distinct TRIGGER component in the components slice, it is silently dropped. A content-equality check against the already-rendered trigger would be safer than a blanket component-type skip.

*Reply to this thread when addressed.*

File: tools/jtk/internal/present/automation_test.go [hunk 5]
```diff
@@ -283,18 +284,85 @@ func TestAutomationPresenter_PresentGetDetail_ShowComponents(t *testing.T) {
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/automation_test.go:284]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): The branch where rule.Trigger == nil and a TRIGGER component is present in rule.Components is not covered by any test. TestComponentTree_TriggerDeduplication covers the trigger != nil case; the nil-trigger code path remains unverified.

*Reply to this thread when addressed.*

File: tools/jtk/internal/present/automation_test.go [hunk 6]
```diff
@@ -283,18 +284,85 @@ func TestAutomationPresenter_PresentGetDetail_ShowComponents(t *testing.T) {
...
 	if !ok {
-		t.Fatalf("expected TableSection at [3], got %T", model.Sections[3])
+		t.Fatalf("expected MessageSection at [3], got %T", model.Sections[3])
 	}
-	if len(table.Rows) != 2 {
-		t.Errorf("expected 2 component rows, got %d", len(table.Rows))
+	if !strings.Contains(tree.Message, "TRIGGER  issue.created") {
+		t.Errorf("expected trigger line, got %q", tree.Message)
+	}
+	if !strings.Contains(tree.Message, "  ACTION  assign.issue") {
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/automation_test.go:299]: 🔵 **Low** (harness-engineering:harness-enforcement-reviewer): PresentGetDetailExtended with showComponents=true has no test after the table-to-tree migration. PresentGetDetail and PresentGetDetailExtended share the same componentTree call, but only the former is exercised; a regression in the extended path (e.g., different section ordering) would go undetected.

*Reply to this thread when addressed.*

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `a5e841e`

> **Approved** with 6 non-blocking suggestions below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-architecture-reviewer | 4 |
| harness-engineering:harness-enforcement-reviewer | 2 |

<details>
<summary><strong>harness-engineering:harness-architecture-reviewer (4 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/api/issues.go:416`
> FormatCustomFieldValue now returns "" for any map[string]any that doesn't match the four known keys and for all unrecognized types. This silently drops field data with no way for users to distinguish an empty field from one whose type is unhandled. The PR intentionally removes noisy map[...] output, but a sentinel like "(unsupported)" or debug-level logging would preserve debuggability when users encounter missing field values.

**💡 Suggestion** - `tools/jtk/api/automation_types.go:82`
> DecodedChildren and DecodedConditions silently swallow JSON unmarshal errors by returning nil. A malformed payload from the Jira API will produce a silently truncated component tree with no signal to the caller or user. Consider returning an error or logging a warning so callers can distinguish 'no children' from 'invalid JSON in children payload'.

**💡 Suggestion** - `tools/jtk/internal/present/automation.go:310`
> componentTree renders all non-TRIGGER entries from the flat components slice at a fixed depth of 1, implicitly assuming they are all direct children of the trigger. If the API ever returns root-level components that are siblings of the trigger (e.g., an error-handler), they will appear incorrectly indented. The assumption is undocumented and untested against real API responses with non-standard component layouts.

**💡 Suggestion** - `tools/jtk/internal/present/automation_test.go:284`
> The branch where rule.Trigger == nil and a TRIGGER component is present in rule.Components is not covered by any test. TestComponentTree_TriggerDeduplication covers the trigger != nil case; the nil-trigger code path remains unverified.

</details>

<details>
<summary><strong>harness-engineering:harness-enforcement-reviewer (2 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/present/automation.go:314`
> TRIGGER deduplication skips all TRIGGER-typed entries in components when rule.Trigger != nil, regardless of whether they represent the same or a different trigger. If a rule legitimately has a distinct TRIGGER component in the components slice, it is silently dropped. A content-equality check against the already-rendered trigger would be safer than a blanket component-type skip.

**💡 Suggestion** - `tools/jtk/internal/present/automation_test.go:299`
> PresentGetDetailExtended with showComponents=true has no test after the table-to-tree migration. PresentGetDetail and PresentGetDetailExtended share the same componentTree call, but only the former is exercised; a regression in the extended path (e.g., different section ordering) would go undetected.

</details>

*4 info-level observations excluded. Run with `--verbose` to include.*

*3 PR discussion threads considered.*

---
<details>
<summary>Completed in 2m 57s | $0.79</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 2m 57s wall · 6m 58s compute (Reviewers: 1m 20s · Synthesis: 1m 34s) |
| Cost | $0.79 |
| Tokens | 207.2k in / 26.9k out |
| Turns | 14 |

</details>

File: tools/jtk/internal/present/automation_test.go (hunk 6 above)
Comment (rianjs) [reply tools/jtk/internal/present/automation_test.go:299]: Validation reply from Codex: confirmed the gh api endpoint with the PR number segment posts an in-thread PR review comment reply.
